### ISSUE TITLE
Preparing deployment incorrectly fails due to check with wrong job spec

### ISSUE THREAD

**Issue by petergtz** [cloudfoundry/bosh#1859]
We recently deployed https://github.com/cloudfoundry/cf-deployment/commit/3f173445ea5044ff3e06d6927bce2080fe164208 which downgrades the loggregator-release from 101.3 to 99 and got the following error message from BOSH:

```
...

Release 'loggregator/99' already exists.

Release 'nats/22' already exists.

Release 'php-buildpack/4.3.44' already exists.

Release 'python-buildpack/1.6.3' already exists.

Release 'routing/0.168.0' already exists.

Release 'nodejs-buildpack/1.6.12' already exists.

Release 'ruby-buildpack/1.7.7' already exists.

Release 'staticfile-buildpack/1.4.20' already exists.

Release 'uaa/53' already exists.

Release 'bits-service/1.4.0-dev.32' already exists.

Release 'statsd-injector/1.0.30' already exists.

  releases:
  - name: loggregator
-   sha1: d9ee90b2ba713f12f2a9a831840a62212ee774fb
+   sha1: 2080e1e0594591dafa716c69f207eb29929bce3d
-   url: https://bosh.io/d/github.com/cloudfoundry/loggregator-release?v=101.3
+   url: https://bosh.io/d/github.com/cloudfoundry/loggregator-release?v=99
-   version: '101.3'
+   version: '99'
Task 5900

Task 5900 | 22:51:46 | Preparing deployment: Preparing deployment (00:00:01)
                     L Error: Job 'metron_agent' in instance group 'consul' specifies link 'doppler', but the release job does not consume it.
Task 5900 | 22:51:47 | Error: Job 'metron_agent' in instance group 'consul' specifies link 'doppler', but the release job does not consume it.

Task 5900 Started  Fri Dec  8 22:51:46 UTC 2017
Task 5900 Finished Fri Dec  8 22:51:47 UTC 2017
Task 5900 Duration 00:00:01
Task 5900 error
Could not deploy: Updating deployment: Expected task '5900' to succeed but state is 'error'
```

Running the following, fixed it for us:
```bash
bosh -e sl delete-release loggregator/101.3
```

However, that shouldn't be necessary, because the check that is happening while `Preparing deployment` should not base that check on the latest available loggregator-release, but on the one specified in the deployment manifest.

**More info:**
The downgrade from 101.3 to 99 removes the following lines from the `metron_agent` job spec:

```
consumes:
- name: doppler
  type: doppler
  optional: true
```

**BOSH director version**: 262.3
**BOSH CPI**: bosh-softlayer-cpi, version 23, but it also happened in aws.

/cc @suhlig @smoser-ibm @idev4u

**Comment by pivotal-jamil-shamy** [comment]
This should be fixed in 262.5 https://<https://github.com/cloudfoundry/bosh/releases/tag/v262.5.0>github.com<https://github.com/cloudfoundry/bosh/releases/tag/v262.5.0>/<https://github.com/cloudfoundry/bosh/releases/tag/v262.5.0>cloudfoundry<https://github.com/cloudfoundry/bosh/releases/tag/v262.5.0>/bosh/releases/tag/<https://github.com/cloudfoundry/bosh/releases/tag/v262.5.0>v262.5.0<https://github.com/cloudfoundry/bosh/releases/tag/v262.5.0>
________________________________

**Comment by petergtz** [comment]
@pivotal-jamil-shamy Thanks for the prompt response! Will update then.
